### PR TITLE
Frontend change

### PR DESCRIPTION
**Pull request by TanayK07**
Styling and Question Page Fixes

### CODE REVIEW

File: src/app/SidebarLayout.tsx [hunk 1]
```diff
@@ -1,3 +1,4 @@
+/* eslint-disable @next/next/no-img-element */
```
Comment (RobinNagpal) [inline src/app/SidebarLayout.tsx:1]: Why is this needed?

File: src/app/courses/view/[...courseInfo]/page.tsx [hunk 2]
```diff
@@ -15,20 +15,24 @@ import Link from 'next/link';
 import { useRouter } from 'next/navigation';
 import { useEffect, useState } from 'react';
 import styled from 'styled-components';
+import { over } from 'lodash';
```
Comment (RobinNagpal) [inline src/app/courses/view/[...courseInfo]/page.tsx:18]: use

`import over from 'lodash/over`

Comment (RobinNagpal) [reply src/app/courses/view/[...courseInfo]/page.tsx:18]: Also is this used anywhere in the code?

File: src/app/courses/view/[...courseInfo]/page.tsx [hunk 3]
```diff
@@ -70,7 +74,7 @@ const CourseView = ({ params, space }: { params: { courseInfo: string[] }; space
     if (!submissionHelper.loadedSubmission) return;
 
     submissionHelper.loadCourseSubmission(courseHelper.course);
-  }, [courseHelper.course, submissionHelper.loadedSubmission]);
+  }, [courseHelper.course, submissionHelper, submissionHelper.loadedSubmission]);
```
Comment (RobinNagpal) [inline src/app/courses/view/[...courseInfo]/page.tsx:77]: Why do we need to add `submissionHelper` as dependency?

File: src/app/SidebarLayout.tsx (hunk 1 above)
Comment (TanayK07) [reply src/app/SidebarLayout.tsx:1]: Done automatically when I ran yarn lint-fix

File: src/app/courses/view/[...courseInfo]/page.tsx [hunk 4]
```diff
@@ -110,7 +114,7 @@ const CourseView = ({ params, space }: { params: { courseInfo: string[] }; space
             )}
           </div>
           <div className="flex flex-col md:flex-row">
-            <StyledNavWrapper className="my-4 relative overflow-scroll border-r-2 h-full">
+            <StyledNavWrapper className="my-4 relative overflow-scroll border-r-2 h">
```
Comment (RobinNagpal) [inline src/app/courses/view/[...courseInfo]/page.tsx:117]: what is `h` class?

File: .env.example [hunk 5]
```diff
@@ -1,20 +0,0 @@
-# Environment variables declared in this file are automatically made available to Prisma.
```
[the remark this answers is not in the document]
Comment (TanayK07) [reply .env.example:1]: I had changed it to .env instead of copying it, I shall push one .env.example right away

File: src/app/courses/view/[...courseInfo]/page.tsx [hunk 6]
```diff
@@ -122,7 +126,8 @@ const CourseView = ({ params, space }: { params: { courseInfo: string[] }; space
                 itemKey={itemKey}
               />
             </StyledNavWrapper>
-            <StyledRightContent className="flex-1 m-4">
+
+            <StyledRightContent className={`flex-1 m-4 ${itemType === 'questions' ? 'overflow-y-hidden' : 'overflow-hidden'}`}>
```
Comment (RobinNagpal) [inline src/app/courses/view/[...courseInfo]/page.tsx:130]: What is the reason to add it questions screen specific class?

File: src/components/app/TreeView/TreeNode.tsx [hunk 7]
```diff
@@ -24,11 +25,41 @@ export const TreeNode: FC<TreeNodeProps> = ({ node, openNodes, setOpenNodes, lev
   };
 
   const isOpen = openNodes[level] === (node.component.key as string);
+  const CheckMark = styled.div`
```
Comment (RobinNagpal) [inline src/components/app/TreeView/TreeNode.tsx:28]: This code doesn't seem to be used. Should we remove it?

File: src/app/courses/view/[...courseInfo]/page.tsx (hunk 3 above)
Comment (TanayK07) [reply src/app/courses/view/[...courseInfo]/page.tsx:77]: Shows Missing dependency fixed by Eslint

File: src/components/courses/Edit/CourseNavigation.tsx [hunk 8]
```diff
@@ -99,11 +111,16 @@ function getReadings(courseKey: string, topic: CourseTopicFragment, readings: Co
...
 }
 
-function getExplanations(courseKey: string, topic: CourseTopicFragment, explanations: CourseExplanationFragment[]) {
+function getExplanations(courseKey: string, topic: CourseTopicFragment, explanations: CourseExplanationFragment[], itemKey: string) {
   return explanations.map((explanation, i) => {
+    const isActive = itemKey === explanation.key;
     return {
       component: (
-        <Link key={explanation.key} className="flex items-center" href={`/courses/view/${courseKey}/${topic.key}/explanations/${explanation.key}`}>
+        <Link
```
Comment (RobinNagpal) [inline src/components/courses/Edit/CourseNavigation.tsx:119]: Do we need the other classes here `flex items-center` ?

File: src/components/courses/Edit/CourseNavigation.tsx [hunk 9]
```diff
@@ -83,12 +84,23 @@ const Container = styled.div`
...
-function getReadings(courseKey: string, topic: CourseTopicFragment, readings: CourseReadingFragment[]) {
+
+function getReadings(courseKey: string, topic: CourseTopicFragment, readings: CourseReadingFragment[], itemKey: string) {
   return readings.map((reading, i) => {
+    const isActive = itemKey === reading.uuid;
     return {
       component: (
-        <Link key={reading.uuid} className="flex items-center" href={`/courses/view/${courseKey}/${topic.key}/readings/${reading.uuid}`}>
+        <Link
+          key={reading.uuid}
```
Comment (RobinNagpal) [inline src/components/courses/Edit/CourseNavigation.tsx:100]: Do we need the other classes here `flex items-center` ?

File: src/components/app/TreeView/TreeNode.tsx (hunk 7 above)
Comment (TanayK07) [reply src/components/app/TreeView/TreeNode.tsx:28]: We can, will be using it in future when I modify the Icon according to click of next button

File: src/app/courses/view/[...courseInfo]/page.tsx (hunk 6 above)
Comment (TanayK07) [reply src/app/courses/view/[...courseInfo]/page.tsx:130]: Because otherwise Questions become scrollable if overflow-hidden is used, makes the UI look bad

File: src/app/courses/view/[...courseInfo]/page.tsx (hunk 4 above)
Comment (TanayK07) [reply src/app/courses/view/[...courseInfo]/page.tsx:117]: Handles the height of the right border on StyledNavWrapper, I increased it to covering the full page instead of just CourseNavigation, as it looks better

File: src/components/courses/View/Details/EvaluationReview.tsx [hunk 10]
```diff
@@ -29,16 +29,45 @@ const TickMark = styled.span`
...
+      props.questionStatus !== 'completed' &&
+      `
+      background-color: transparent;
+      border: 2px solid #D32F2F;
+    `}
   }
 `;
 
+const QuestionNumber = styled.div`
+  background-color: #d32f2f;
```
Comment (RobinNagpal) [inline src/components/courses/View/Details/EvaluationReview.tsx:59]: For color and bacground color we use variables.

```
  background-color: var(--bg-color);
  color: var(--text-color);
```
We can use multiple themes in our website

File: src/components/courses/Edit/CourseNavigation.tsx (hunk 8 above)
Comment (TanayK07) [reply src/components/courses/Edit/CourseNavigation.tsx:119]: `flex items-center` was already there, ` ${isActive ? 'underline' : ''}` underlines/highlights the current Selected Node in tree

File: src/components/courses/Edit/CourseNavigation.tsx (hunk 9 above)
Comment (TanayK07) [reply src/components/courses/Edit/CourseNavigation.tsx:100]: `flex items-center` was already there, ` ${isActive ? 'underline' : ''}` underlines/highlights the current Selected Node in tree

File: src/components/courses/View/Details/EvaluationReview.tsx (hunk 10 above)
Comment (TanayK07) [reply src/components/courses/View/Details/EvaluationReview.tsx:59]: Okay, I can do that , but that particular shade of red picked up from the previous website is only used there and nowhere else in the website

File: src/components/courses/Edit/CourseNavigation.tsx [hunk 11]
```diff
@@ -234,9 +270,10 @@ const CourseComponent: React.FC<CourseNavigationProps> = ({ course, showAddModal
         '1': topicKey! + '_chapter_root',
       });
     }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (RobinNagpal) [inline src/components/courses/Edit/CourseNavigation.tsx:273]: Remove this line
